splinepy/io/npz.py

```diff
@@ -10,60 +10,108 @@
+        if key.startswith("spline_") and key not in processed_keys:
+            # Extract the index of the spline from the key
+            i = int(key.split("_")[1])
+
+            # Initialize an empty dictionary to store the properties of the spline
+            dict_spline = {}
+
+            # Add the common properties of all splines
+            dict_spline["control_points"] = loaded[
+                f"spline_{i}_control_points"
+            ]
+            dict_spline["degrees"] = loaded[f"spline_{i}_degrees"]
+
+            # Add the weights if the spline is rational
+            if f"spline_{i}_weights" in loaded:
+                dict_spline["weights"] = loaded[f"spline_{i}_weights"]
 
-    # update the rest
-    dict_spline.update(control_points=loaded["control_points"])
-    dict_spline.update(degrees=loaded["degrees"])
-    dict_spline.update(knot_vectors=eval(loaded["knot_vectors"][0]))
+            # Add the knot vectors if the spline has them
+            if f"spline_{i}_knot_vectors_0" in loaded:
+                kvs = []
+                for j in range(dict_spline["degrees"].size):
+                    kvs.append(loaded[f"spline_{i}_knot_vectors_{j}"])
+                dict_spline["knot_vectors"] = kvs
 
-    return dict_spline
+            # Append dictionary of spline to the list
+            list_of_spline_dicts.append(dict_spline)
 
+            # Add keys to processed_keys set
+            processed_keys.add(f"spline_{i}_control_points")
+            processed_keys.add(f"spline_{i}_degrees")
+            processed_keys.add(f"spline_{i}_weights")
+            processed_keys.add(f"spline_{i}_whatami")
+            for j in range(dict_spline["degrees"].size):
+                processed_keys.add(f"spline_{i}_knot_vectors_{j}")
 
-def export(fname, spline):
+    return io.ioutils.dict_to_spline(list_of_spline_dicts)
+
+
+def export(fname, list_of_splines):
     """
-    Save spline as `.npz`.
+    Save a list of splines as `.npz`.
 
     Parameters
     -----------
     fname: str
-    spline: BSpline or NURBS
+    splines: list of NURBS / BSpline / Bezier / Rational Bezier
 
     Returns
     --------
     None
     """
-    property_dicts = {
-        "degrees": spline.degrees,
-        "knot_vectors": np.array([str(spline.knot_vectors)]),
-        "control_points": spline.control_points,
-        "whatami": np.array([spline.whatami]),
-    }
+    # Initialize an empty dictionary to store the properties of each spline
+    property_dicts = {}
+
+    # Loop through the list of splines and add their properties to the dictionary
+    for i, spline in enumerate(list_of_splines):
+        # Use a prefix to distinguish different splines
+        prefix = f"spline_{i}_"
+
+        # Add the common properties of all splines
+        property_dicts[f"{prefix}degrees"] = spline.degrees
+        property_dicts[f"{prefix}control_points"] = spline.control_points
+        property_dicts[f"{prefix}whatami"] = np.array([spline.whatami])
```

Comment:
```suggestion
```
we don't use this information for now

Reply:
Fixed in [08105e1](https://github.com/tataratat/splinepy/pull/280/commits/08105e1163ee942d3dc2d9e6c02498f0fa925718)